### PR TITLE
[AutoDiff] Gardening.

### PR DESCRIPTION
**Pull request by dan-zheng**
- Fix `AdjointEmitter` doc comments to use accumulation when appropriate.
- Remove unnecessary newline from two debug logs.
- Minor doc comment fixes/edits.

### CODE REVIEW

File: lib/SILOptimizer/Mandatory/Differentiation.cpp [hunk 1]
```diff
@@ -3089,8 +3089,7 @@ class Cleanup {
   void apply(SILBuilder &builder, SILLocation loc) {
     if (!func) return;
     assert(value);
-    LLVM_DEBUG(getADDebugStream() << "Running `Cleanup::apply` for "
-               << value << '\n');
+    LLVM_DEBUG(getADDebugStream() << "Running `Cleanup::apply` for " << value);
```
Comment (dan-zheng) [inline lib/SILOptimizer/Mandatory/Differentiation.cpp]: Regarding debug log changes:
* Old:
```
[AD] Running `Cleanup::apply` for   %7 = alloc_stack $τ_0_0.CotangentVector        // users: %56, %41, %8

[AD] Cleaning up local buffer   %7 = alloc_stack $τ_0_0.CotangentVector        // users: %56, %41, %8

[AD] Running `Cleanup::apply` for   %4 = alloc_stack $τ_0_0.CotangentVector        // users: %57, %5

[AD] Cleaning up local buffer   %4 = alloc_stack $τ_0_0.CotangentVector        // users: %57, %5

[AD] Generated adjoint
```

* New:
```
[AD] Running `Cleanup::apply` for   %7 = alloc_stack $τ_0_0.CotangentVector        // users: %56, %41, %8
[AD] Cleaning up local buffer   %7 = alloc_stack $τ_0_0.CotangentVector        // users: %56, %41, %8
[AD] Running `Cleanup::apply` for   %4 = alloc_stack $τ_0_0.CotangentVector        // users: %57, %5
[AD] Cleaning up local buffer   %4 = alloc_stack $τ_0_0.CotangentVector        // users: %57, %5
[AD] Generated adjoint.
```

It seems the printed `SILValue` always has a newline at the end - verified via visual inspection based on existing tests.

**Comment by dan-zheng** [comment]
@swift-ci Please test tensorflow

File: lib/SILOptimizer/Mandatory/Differentiation.cpp (hunk 1 above)
Comment (rxwei) [reply lib/SILOptimizer/Mandatory/Differentiation.cpp]: Not sure if that's the case if it's a SIL argument.

**Review by rxwei — APPROVED** [review]

**Comment by dan-zheng** [comment]
> Not sure if that's the case if it's a SIL argument.

True, though I don't think we'd ever clean up arguments. Could you confirm?

**Comment by dan-zheng** [comment]
macOS CI flaked, merging.

**Pull request merged**